CIP-0030/README.md

```diff
@@ -158,57 +187,65 @@ type TxSignError = {
 * ProofGeneration - User has accepted the transaction sign, but the wallet was unable to sign the transaction (e.g. not having some of the private keys)
 * UserDeclined - User declined to sign the transaction
 
-
-
-## Initial API
+### Initial API
 
 In order to initiate communication from webpages to a user's Cardano wallet, the wallet must provide the following javascript API to the webpage. A shared, namespaced `cardano` object must be injected into the page if it did not exist already. Each wallet implementing this standard must then create a field in this object with a name unique to each wallet containing a `wallet` object with the following methods. The API is split into two stages to maintain the user's privacy, as the user will have to consent to `cardano.{walletName}.enable()` in order for the dApp to read any information pertaining to the user's wallet with `{walletName}` corresponding to the wallet's namespaced name of its choice.
 
-### cardano.{walletName}.enable(): Promise\<API>
+#### cardano.{walletName}.enable(extensions: Extension[] = []): Promise\<API>
```

Comment:
Should `Extension` just be a string? I takes perhaps away the ability to pass more information upon requesting a new extension, which can be useful for the user to make a decision to enable it or not. For example in #296 `cardano.{walletName}.governance.enable(purpose: VotingPurpose[]): Promise\<API>`

Reply:
Not sure to follow you? What would be the structure of that extension string?

Reply:
@KtorZ I think ehanoc is asking about/suggesting something like

```
cardano.{walletName}.enable([{ cip: "CIP-0062", parameters: { purpose: ["catalyst", "sundae"] } }]);
```

I like the simplicity of the string approach, since we could balloon forever on complexity here (enable by patterns, versioning, fallbacks, etc.)

That being said, I can see the UX risk if the alternative is to prompt the user twice: once to enable an extension, and another to configure that extension.

In this case, perhaps it would be sufficient for each voting purpose to be a separate CIP that you could request, but it's only one step further where that starts to break down; Imagine a hypothetical CIP that allows wallets to sign transactions without a prompt (for instant trade experiences):

```
cardano.walletName.enable({ cip: "CIP-XXXX", spendingLimit: { coins: 100000000, assets: {...} } })
```